src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -4153,6 +4153,36 @@ private static async Task<(string version, string hash)> ParseLatestVersionFile(
             }
         }
 
+        /// <summary>
+        /// Parses files that contains lines with [prefix]:[sha], [version]
+        /// </summary>
+        private static async Task<(string version, string hash)> ParseVersionsFile(string urlOrFilename, string prefix)
+        {
+            var content = urlOrFilename.StartsWith("http", StringComparison.OrdinalIgnoreCase)
+                ? await DownloadContentAsync(urlOrFilename)
+                : await File.ReadAllTextAsync(urlOrFilename)
+                ;
+
+            using (var sr = new StringReader(content))
+            {
+                string line = null;
+
+                while (null != (line = sr.ReadLine()))
```

Comment:
Don't you want this to fail loudly if something is unexpected in the format? e.g. if there are no lines with the prefix or something.